### PR TITLE
DM-7421: Get processFile working again

### PR DESCRIPTION
**Pull request by timj**

### CODE REVIEW

File: bin.src/processFile.py [hunk 1]
```diff
@@ -323,12 +328,13 @@ def __iter__(self):
                         help="display the specified information to stdout and quit (unless run is specified).")
     parser.add_argument("-L", "--loglevel", help="logging level", default="WARN",
                         choices=CaseInsensitiveChoices('DEBUG', 'INFO', 'WARN', 'FATAL'))
-    parser.add_argument("-T", "--trace", nargs="*", action=pbArgparse.TraceLevelAction,
-                        help="trace level for component", metavar="COMPONENT=LEVEL")
+    if False:
```
Comment (wmwv) [inline bin.src/processFile.py]: Add a comment explaining
(a) why the `--trace` option is currently disable
(b) when it should be reactivated

File: bin.src/processFile.py [hunk 2]
```diff
@@ -140,8 +143,8 @@ def run(config, inputFiles, weightFiles=None, varianceFiles=None,
         # process the data
         #
         if config.doCalibrate:
-            result = calibrateTask.run(exposure)
-            exposure, calibSources = result.exposure, result.sources
+            result = charImageTask.characterize(exposure)
+            exposure, calibSources = result.exposure, result.sourceCat
         else:
             calibSources = None
             if not exposure.getPsf():
```
Comment (wmwv) [inline bin.src/processFile.py:150]: The line just after this (Line 151) needs to be updated to `charImageTask`

```
            if not exposure.getPsf():
                calibrateTask.installInitialPsf(exposure)
```

should presumably be

```
            if not exposure.getPsf():
                charImageTask.installInitialPsf(exposure)
```

File: bin.src/processFile.py [hunk 3]
```diff
@@ -323,12 +328,13 @@ def __iter__(self):
...
-    parser.add_argument("-T", "--trace", nargs="*", action=pbArgparse.TraceLevelAction,
-                        help="trace level for component", metavar="COMPONENT=LEVEL")
+    if False:
+        parser.add_argument("-T", "--trace", nargs="*", action=pbArgparse.TraceLevelAction,
+                            help="trace level for component", metavar="COMPONENT=LEVEL")
 
     parser.add_argument('--debug', '-d', action="store_true", help="Load debug.py?", default=False)
     parser.add_argument('--display', nargs='*', help="Display sources on an image display",
-                        choices=CaseInsensitiveChoices('True', 'showApertures', "showShapes"),
+                        choices=CaseInsensitiveChoices('True', 'showApertures', "showShapes", "showPSFs"),
```
Comment (wmwv) [inline bin.src/processFile.py:333]: Add 'showPSFs' option to documentation in `doc/main.dox`.

Actually, now that I look at it more closely, update documentation in `doc/main.dox` from `--ds9` to talk about the more general `--display` option.

File: bin.src/processFile.py (hunk 1 above)
Comment (RobertLuptonTheGood) [reply bin.src/processFile.py]: Removed (Trace support appears to be gone from pipe_base)

File: bin.src/processFile.py (hunk 2 above)
Comment (RobertLuptonTheGood) [reply bin.src/processFile.py:150]: Good catch, but the API's changed a bit.  Fixed.

**Comment by wmwv** [comment]
Looks good.  Line-by-line comments in PR.  Would also be great to run a pass through to strip trailing whitespace (there are several changes that are purely from unintentionally adding 4 spaces to a blank line, likely courtesy of an over-helpful editor.)

**Pull request merged**